e2e/test/scenarios/collections/collections.cy.spec.js

```diff
@@ -718,6 +721,72 @@ describe("scenarios > collection defaults", () => {
             );
           });
         });
+
+        it("moving collections should disable moving into any of the moving collections in recents or search (metabase#45248)", () => {
+          createCollection({ name: "Outer collection 1" }).then(
+            ({ body: { id: parentCollectionId } }) => {
+              cy.wrap(parentCollectionId).as("outerCollectionId");
+              createCollection({
+                name: "Inner collection 1",
+                parent_id: parentCollectionId,
+              }).then(({ body: { id: innerCollectionId } }) => {
+                cy.wrap(innerCollectionId).as("innerCollectionId");
+              });
+              createCollection({
+                name: "Inner collection 2",
+                parent_id: parentCollectionId,
+              });
+            },
+          );
```

Comment:
fwiw - and you may know this and perhaps just avoided this pattern for a good reason - but I recently learned you can avoid nested `then`s with `this`. For example:

```typescript
createCollection({ name: "Outer collection 1" }).then((response) =>
  cy.wrap(response.body.id).as("parentCollectionId")
);

cy.then(function () {
  createCollection({
    name: "Inner collection 1",
    parent_id: this.parentCollectionId,
  }).then((response) => {
    cy.wrap(response.body.id).as("innerCollectionId");
  });
});

cy.then(function () {
  createCollection({
    name: "Inner collection 2",
    parent_id: this.parentCollectionId,
  });
});
```

Thinking out loud: To make it easier to retrieve a collection's id after creating it, it might be nice to add a `wrapId` and `idAlias` parameter to `createCollection`, like we have in `createQuestion` (in `e2e/support/helpers/e2e-question-helpers`).

Reply:
doesn't this have the same amount of nesting?

Reply:
Yes! My bad. Still, kind of a nice pattern for readability I think